File: 2025/day10/python/debug.py
#!/usr/bin/env python3

from solution import parse_line_part2, solve_machine_part2
from pathlib import Path

input_file = Path(__file__).parent.parent / 'input.txt'
lines = input_file.read_text().strip().split('\n')

print("Testing first 5 machines:")
for i, line in enumerate(lines[:5], 1):
    n_counters, joltage, buttons = parse_line_part2(line)
    result = solve_machine_part2(n_counters, joltage, buttons)
    print(f"Machine {i}: {result}")

print("\nTotal for first 5:")
print(sum(solve_machine_part2(*parse_line_part2(line)) for line in lines[:5]))

File: 2025/day10/clojure/debug.clj
#!/usr/bin/env clojure

(load-file "solution.clj")

(def input (slurp "../input.txt"))
(def lines (clojure.string/split-lines input))

(println "Testing first 5 machines:")
(doseq [[i line] (map-indexed vector (take 5 lines))]
  (let [[n-counters joltage buttons] (day10.solution/parse-line-part2 line)
        result (day10.solution/solve-machine-part2 n-counters joltage buttons)]
    (println (str "Machine " (inc i) ": " result))))

(println "\nTotal for first 5:")
(println (reduce + (map (fn [line]
                          (let [[n-counters joltage buttons] (day10.solution/parse-line-part2 line)]
                            (day10.solution/solve-machine-part2 n-counters joltage buttons)))
                        (take 5 lines))))
